[bitnami/kiam] Add kiam helm chart (#4313)
bitnami/charts: bitnami/kiam/README.md

| `server.podSecurityContext`                 | Pod security context                                                                        | `{}`                                     |
| `server.assumeRoleArn`                      | IAM role for the server to assume                                                           | `nil`                                    |
| `server.sessionDuration`                    | Session duration for STS tokens                                                             | `15m`                                    |
| `server.useHostNetwork`                     | Use host networking (ports will be directly exposed in the host)                            | `false`                                  |
| `server.tlsCerts.certFileName`              | Name of the certificate filename                                                            | `cert.pem`                               |
| `server.tlsCerts.keyFileName`               | Name of the certificate filename                                                            | `key.pem`                                |

> | `server.tlsCerts.caFileName`                | Name of the certificate filename                                                            | `ca.pem`                                 |

Author: After checking, the changes would make it a bit more complicated as it is using `toYaml` in the template code. I'd rather leave it as it is.

| `server.initContainers`                     | Add additional init containers to the kiam pods                                             | `{}` (evaluated as a template)           |
| `server.sidecars`                           | Add additional sidecar containers to the kiam pods                                          | `{}` (evaluated as a template)           |

### kiam agent parameters

| Parameter                                   | Description                                                                                | Default                                  |
|---------------------------------------------|--------------------------------------------------------------------------------------------|------------------------------------------|
| `agent.enabled`                             | Deploy the kiam agent                                                                      | `true`                                   |
| `agent.containerPort`                       | HTTPS port to expose at container level                                                    | `8443`                                   |

> | `agent.allowRouteRegexp`                    | Regexp with the allowed paths for agents to redirect                                       | `nil`                                    |

Reviewer: I would suggest `allowedRoutesRegExp` since it looks more appropriate for what it supposedly does

Author: The KIAM flag is like this
```
--allow-route-regexp
```

For consistency with the application, I'd leave the same name, but I will use RegExp instead of Regexp

> | `agent.iptables`                            | Have the agent modify the host iptables rules                                              | `false`                                  |

Reviewer: This option name is a bit weird. Maybe `modifyIptables` is a better choice of words? We should at least try to use something more descriptive of what it actually does.

Author: Once again, it is because of the KIAM flag

```
--iptables
```

Reviewer: ok

| `agent.iptablesRemoveOnShutdown`            | Remove iptables rules when shutting down the agent node                                    | `false`                                  |
| `agent.hostInterface`                       | Interface for agents for redirecting requests                                              | `cali+`                                  |
| `agent.logJsonOutput`                       | Use JSON format for logs                                                                   | `true`                                   |
| `agent.keepaliveParams.time`                | Keepalive time                                                                             | `nil`                                    |
| `agent.keepaliveParams.timeout`             | Keepalive timeout                                                                          | `nil`                                    |
| `agent.keepaliveParams.permitWithoutStream` | Permit keepalive without stream                                                            | `nil`                                    |

> | `agent.deepProbe`                           | Use the probes using the `/health` endpoint                                                | `false`                                  |

Reviewer: I would suggest `enableDeepProbe`

Author: Done

Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,

```bash
helm install my-release --set server.useDeployment=true bitnami/kiam
```

The above command sets the server nodes to be deployed as Deployment objects.

Alternatively, a YAML file that specifies the values for the parameters can be provided while installing the chart. For example,

```bash
$ helm install my-release -f values.yaml bitnami/kiam
```

> **Tip**: You can use the default [values.yaml](values.yaml)

## Configuration and installation details

### [Rolling VS Immutable tags](https://docs.bitnami.com/containers/how-to/understand-rolling-tags-containers/)

It is strongly recommended to use immutable tags in a production environment. This ensures your deployment does not change automatically if the same tag is updated with a different image.

Bitnami will release a new chart updating its containers if a new version of the main container, significant changes, or critical vulnerabilities exist.

> Note also that if you disable PostgreSQL per above you MUST supply values for the `externalDatabase` connection.

Reviewer: ```suggestion
Also, note that if you disable PostgreSQL as per above, you MUST supply values for the `externalDatabase` connection.
```

Author: This should not be here

### Setting Pod's affinity

This chart allows you to set your custom affinity using the `server.affinity` and `agent.affinity` paremeters. Find more infomation about Pod's affinity in the [kubernetes documentation](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity).

As an alternative, you can use of the preset configurations for pod affinity, pod anti-affinity, and node affinity available at the [bitnami/common](https://github.com/bitnami/charts/tree/master/bitnami/common#affinities) chart. To do so, set the `server.podAffinityPreset`, `agent.podAffinityPreset`, `server.podAntiAffinityPreset`, `agent.podAntiAffinityPreset`, or `server.nodeAffinityPreset` and `agent.nodeAffinityPreset` parameters.

### TLS Secrets

This chart will facilitate the creation of TLS secrets for use with kiam. There are three common use cases:

- Helm auto-generates the certificates.
- User specifies the certificates in the values.
- User generates/manages certificates separately.

By default the first use case will be applied. In second case, it's needed a certificate and a key. We would expect them to look like this:

> - certificate files should look like (and there can be more than one certificate if there is a certificate chain)

Reviewer: ```suggestion
- The certificate files (there can be more than one certificate if there is a certificate chain) should look like this: 
```

Author: Done

    ```console
    -----BEGIN CERTIFICATE-----
    MIID6TCCAtGgAwIBAgIJAIaCwivkeB5EMA0GCSqGSIb3DQEBCwUAMFYxCzAJBgNV
    ...
    jScrvkiBO65F46KioCL9h5tDvomdU1aqpI/CBzhvZn1c0ZTf87tGQR8NK7v7
    -----END CERTIFICATE-----
    ```

> - keys should look like:

Reviewer: ```suggestion
- The keys should look like this:
```

Reviewer: don't forget this
